core/src/main/scala/za/co/absa/spline/core/harvester/componentCreators.scala

```diff
@@ -110,7 +111,7 @@ object ExpressionConverter {
     })
   }
 
-  private[this] def renderValue(recursion: Any => Option[Any])(o: Any): Option[Any] = {
+  private[this] def renderValue(recursion: Any => Option[Any], t : Option[org.apache.spark.sql.types.DataType] = None)(o: Any): Option[Any] = {
```

Comment:
As `t` describes the type of `o` they both should logically be in the same parameter group.